File: src/python/flask/app/routes/cache.py
from flask import Blueprint, request, jsonify, current_app
from app.services.cache import get_cache_service
import time
from datetime import datetime

bp = Blueprint('cache', __name__)

# The TTL is part of the response contract and must match what is written
# to Redis. See contracts/rest/canonical-payloads.md.
CACHE_TTL_SECONDS = 300


@bp.route('/cache')
def cache_endpoint():
    """Cache operations endpoint"""
    try:
        key = request.args.get('key', default='test')
        if not key:
            return jsonify({
                'error': 'Key parameter is required'
            }), 400

        cache_service = get_cache_service()

        # Try to get from cache first
        cached_value = cache_service.get(key)

        if cached_value:
            return jsonify({
                'key': key,
                'value': cached_value,
                'cached': True,
                'ttl': CACHE_TTL_SECONDS,
                'timestamp': datetime.utcnow().isoformat()
            })

        # If not in cache, create new value
        value = f"Cached value for {key} at {datetime.utcnow().isoformat()}"

        # Store in cache with 5 minute TTL
        cache_service.set(key, value, ttl=CACHE_TTL_SECONDS)

        return jsonify({
            'key': key,
            'value': value,
            'cached': False,
            'ttl': CACHE_TTL_SECONDS,
            'timestamp': datetime.utcnow().isoformat()
        })

    except Exception as e:
        current_app.logger.error(f"Cache operation failed: {str(e)}")
        return jsonify({
            'error': 'Cache operation failed'
        }), 500

File: src/bun/hono/src/routes/cache.ts
import { Hono } from 'hono';
import { cacheService } from '../services/cache.ts';
import type { CacheResponse } from '../types.ts';

// The TTL is part of the response contract and must match what is written
// to Redis. See contracts/rest/canonical-payloads.md.
const CACHE_TTL_SECONDS = 300;

export const cacheRoutes = new Hono();

// Cache endpoint
cacheRoutes.get('/cache', async (c) => {
  try {
    const key = c.req.query('key') || 'test';
    const cached = await cacheService.get(key);

    if (cached !== null) {
      const response: CacheResponse = {
        key,
        value: cached,
        cached: true,
        ttl: CACHE_TTL_SECONDS,
      };

      return c.json(response);
    }

    const value = `cached_data_${key}_${Date.now()}`;
    await cacheService.set(key, value, CACHE_TTL_SECONDS);

    const response: CacheResponse = {
      key,
      value,
      cached: false,
      ttl: CACHE_TTL_SECONDS,
    };

    return c.json(response);
  } catch (error) {
    return c.json({
      error: 'Internal Server Error',
      message: 'An error occurred while processing the request',
    }, 500);
  }
});
